File: Examples/CSharp/AsposePDF/Links-Actions/UpdateLinks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures.Annotations;
using Aspose.Pdf.InteractiveFeatures;

namespace CSharp.AsposePdf.LinksActions
{
    public class UpdateLinks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_LinksActions();

            // Load the PDF file
            Document doc = new Document(dataDir+ "UpdateLinks.pdf");
            // Get the first link annotation from first page of document
            LinkAnnotation linkAnnot = (LinkAnnotation)doc.Pages[1].Annotations[1];
            // Modification link: change link destination
            GoToAction goToAction = (GoToAction)linkAnnot.Action;
            // Specify the destination for link object
            // The first parameter is document object, second is destination page number.
            // The 5ht argument is zoom factor when displaying the respective page. When using 2, the page will be displayed in 200% zoom
            goToAction.Destination = new Aspose.Pdf.InteractiveFeatures.XYZExplicitDestination(doc, 1, 1, 2, 2);
            // Save the document with updated link
            doc.Save(dataDir + "PDFLINK_Modified_UpdateLinks_out.pdf");

            
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Links-Actions/UpdateLinks.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports Aspose.Pdf.Annotations
Namespace AsposePDF.LinksActions
    Public Class UpdateLinks
        Public Shared Sub Run()
            Try
                ' ExStart:UpdateLinks
                ' The path to the documents directory.
                Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_LinksActions()
                ' Load the PDF file
                Dim doc As New Document(dataDir & Convert.ToString("UpdateLinks.pdf"))
                ' Get the first link annotation from first page of document
                Dim linkAnnot As LinkAnnotation = DirectCast(doc.Pages(1).Annotations(1), LinkAnnotation)
                ' Modification link: change link destination
                Dim goToAction As GoToAction = DirectCast(linkAnnot.Action, GoToAction)
                ' Specify the destination for link object
                ' The first parameter is document object, second is destination page number.
                ' The 5ht argument is zoom factor when displaying the respective page. When using 2, the page will be displayed in 200% zoom
                goToAction.Destination = New XYZExplicitDestination(1, 1, 2, 2)
                dataDir = dataDir & Convert.ToString("PDFLINK_Modified_UpdateLinks_out_.pdf")
                ' Save the document with updated link
                doc.Save(dataDir)
                ' ExEnd:UpdateLinks
                Console.WriteLine(Convert.ToString(vbLf & "Links updated successfully." & vbLf & "File saved at ") & dataDir)
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
        End Sub
    End Class
End Namespace